### PR TITLE
Glt 990 slow pools by library

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

**Comment by callunity** [comment]
@apmasell , the numbering is all out of whack...

**Comment by apmasell** [comment]
Fixed.

**Comment by tonydebat** [comment]
After merging, unit test fails with `mvn clean install -P external`.

> Caused by: org.flywaydb.core.api.FlywayException: Schema "public" contains a failed migration to version 0033 !
> 	at org.flywaydb.core.internal.command.DbMigrate$2.doInTransaction(DbMigrate.java:196)

**Comment by apmasell** [comment]
Sorry, I was lack a test block for the primary key changes.

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLLibraryDAO.java [hunk 1]
```diff
@@ -127,6 +127,16 @@
 
   public static final String LIBRARY_SELECT_BY_BOX_POSITION_ID = LIBRARIES_SELECT + " WHERE l.boxPositionId = ?";
 
+  private static final String LIBRARY_SELECT_BY_ADJACENT = LIBRARIES_SELECT + " JOIN Sample ON Library.sample_sampleId = Sample.sampleId, "
+      + "Library TargetLibrary JOIN Sample TargetSample ON TargetLibrary.sample_sampleId = TargetSample.sampleId WHERE TargetLibrary.libraryId = ? "
```
Comment (callunity) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLLibraryDAO.java]: I'm getting an error in the editLibrary page: `nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'Library.libraryId' in 'where clause'`

Comment (apmasell) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLLibraryDAO.java]: Fixed. Sorry. I made a mess of this code.

**Comment by callunity** [comment]
:+1: :100: Edit Library page is super speedy, and List X pages are all down to 5 seconds.

**Comment by tonydebat** [comment]
👍

**Pull request merged**